### PR TITLE
Simplify diagonal computation

### PR DESCRIPTION
**Pull request by DavidSCN**

### CODE REVIEW

File: include/mf_nh_operator.h [hunk 1]
```diff
@@ -1210,34 +1180,23 @@ NeoHookOperator<dim, fe_degree, n_q_points_1d, Number>::compute_diagonal()
...
+                                    &NeoHookOperator::do_operation_on_cell,
+                                    this);
 
+  data_in_use->initialize_dof_vector(inverse_diagonal_vector);
   // set_constrained_entries_to_one
   Assert(data_current->get_constrained_dofs().size() ==
            data_reference->get_constrained_dofs().size(),
          ExcInternalError());
   for (const auto dof : data_current->get_constrained_dofs())
     diagonal_vector.local_element(dof) = 1.;
```
Comment (DavidSCN) [inline include/mf_nh_operator.h:1194]: @peterrum this is still required, right?
BTW your tool implementation is more than twice as slow as the implementation here. This was a bit unexpected.

Comment (peterrum) [reply include/mf_nh_operator.h:1194]: > this is still required, right?

Yes. Thos values are undefined (set to zero).

> BTW your tool implementation is more than twice as slow as the implementation here. This was a bit unexpected.

I know. This is the cost for correctness and explicit operations on lanes (like  https://github.com/dealii/dealii/blob/c0b01e2791e99632132e68c6f78fa44c57d79d9e/include/deal.II/matrix_free/tools.h#L541-L578 ) ;)

Comment (peterrum) [reply include/mf_nh_operator.h:1194]: I guess one could optimize the code for standard containment cases...

Comment (DavidSCN) [reply include/mf_nh_operator.h:1194]: Alright, thanks!

Comment (DavidSCN) [reply include/mf_nh_operator.h:1194]: Maybe `vectorization_category_strict` would help to bypass the explicit code snippet. I only have two types of constraints here (DRB only one).

**Pull request merged**